scripts/travis_target.sh

```diff
@@ -1,84 +1,72 @@
+    if [[ ! "$line" =~ ^\(maint|doc|packaging|fact-[0-9]+\)|revert ]]; then
+      echo -e \
+          "\n\n\n\tThis commit summary didn't match CONTRIBUTING.md guidelines:\n" \
+          "\n\t\t$line\n" \
+          "\tThe commit summary (i.e. the first line of the commit message) should start with one of:\n" \
+          "\t\t(FACT-<digits>) # this is most common and should be a ticket at tickets.puppetlabs.com\n" \
+          "\t\t(doc)\n" \
+          "\t\t(maint)\n" \
+          "\t\t(packaging)\n" \
+          "\t\tRevert\n" \
+          "\n\tThis test for the commit summary is case-insensitive.\n\n\n"
+      exit 1
     fi
+  done
+  exit 0
+elif [ ${TRAVIS_TARGET} == DEBUG ]; then
+  cmake -DCMAKE_BUILD_TYPE=Debug -DCOVERALLS=ON .
+else
+  cmake .
+fi
 
-    # Build facter
-    if [ $1 == "cpplint" ]; then
-        export MAKE_TARGET="cpplint"
-    elif [ $1 == "cppcheck" ]; then
-        export MAKE_TARGET="cppcheck"
-    else
-        export MAKE_TARGET="all"
-    fi
-    make $MAKE_TARGET -j2
-    if [ $? -ne 0 ]; then
-        echo "build failed."
-        exit 1
-    fi
+if [ ${TRAVIS_TARGET} == CPPLINT ]; then
+  make cpplint
+elif [ ${TRAVIS_TARGET} == DOXYGEN ]; then
+  # Build docs
+  pushd lib
+  doxygen 2>&1 | ( ! grep . )
+  ruby docs/generate.rb > /tmp/facts.md
+  popd
+elif [ ${TRAVIS_TARGET} == CPPCHECK ]; then
+  make cppcheck
+else
+  make -j2
 
-    # Run tests if not doing cpplint/cppcheck
-    if [ $1 == "cpplint" ]; then
-        # Verify documentation
-        pushd ../lib
-        doxygen 2>&1 | ( ! grep . )
-        if [ $? -ne 0 ]; then
-            echo "documentation failed."
-            exit 1
-        fi
-        ruby docs/generate.rb > /tmp/facts.md
-        if [ $? -ne 0 ]; then
-            echo "fact documentation failed."
-            exit 1
-        fi
-        popd
-    elif [ $1 != "cppcheck" ]; then
-        # Install the bundle before testing
-        bundle install --gemfile=../lib/Gemfile
-        if [ $? -ne 0 ]; then
-            echo "bundle install failed."
-            exit 1
-        fi
+  # Install the bundle before testing
+  bundle install --gemfile=lib/Gemfile
+  make test ARGS=-V
 
-        LD_PRELOAD=/lib/x86_64-linux-gnu/libSegFault.so ctest -V 2>&1 | c++filt
-        if [ ${PIPESTATUS[0]} -ne 0 ]; then
-            echo "tests reported an error."
-            exit 1
-        fi
+  # Make sure installation succeeds
+  make DESTDIR=$USERDIR install
 
-        if [ $1 == "debug" ]; then
-            coveralls --gcov gcov-4.8 --gcov-options '\-lp' -r .. >/dev/null
-        fi
-
-        # Make sure installation succeeds
-        make DESTDIR=$USERDIR install
-        if [ $? -ne 0 ]; then
-            echo "install failed."
-            exit 1
-        fi
-    fi
-}
+  if [ ${TRAVIS_TARGET} == DEBUG ]; then
+    # Ignore coveralls failures, keep service success uncoupled
+    coveralls --gcov gcov-4.8 --gcov-options '\-lp' -e vendor >/dev/null || true
```

Comment:
Moving the build dir to root includes files that aren't compiled on this platform. It's a more complete view of what's tested in Travis, but doesn't include testing from Windows.